web-app/src/components/BookingModal/index.js

```diff
@@ -1,102 +1,118 @@
 import React from 'react';
-import DatePicker from 'react-datepicker';
 import { PropTypes as PT } from 'prop-types';
-import { Button, Modal } from '../common';
+import { Button, Modal, Input } from '../common';
 import FontAwesomeIcon from '@fortawesome/react-fontawesome';
 import { faTimes } from '@fortawesome/fontawesome-free-solid';
-import { StyleContainer, BookingStatus, BookingInputs, StatusDot, ButtonWrap } from './styled';
+import {
+  StyleContainer,
+  BookingStatus,
+  StatusDot,
+  ButtonWrap,
+  Form,
+} from './styled';
 import { statusText } from '../../utilities/holidayStatus';
 
 const BookingModal = props => {
   const {
     closeModal,
     showModal,
     booking,
-    changeStart,
-    changeEnd,
-    changeHalfday,
+    formElementsArray,
     requestHoliday,
     cancelHoliday,
     updateHoliday,
-    changeWFH,
+    formChanged,
   } = props;
 
-  const getActions = () => (
-    booking.title ?
+  const getActions = () =>
+    booking.title ? (
       <ButtonWrap>
-        <Button id="updateHolidayBtn" onClick={updateHoliday} label={'Update'} />
-        <Button id="cancelHolidayBtn" onClick={cancelHoliday} label={'Cancel Booking'} />
+        <Button
+          id="updateHolidayBtn"
+          onClick={updateHoliday}
+          label={'Update'}
+          disabled={!booking.formIsValid}
+        />
+        <Button
+          id="cancelHolidayBtn"
+          onClick={cancelHoliday}
+          label={'Cancel Booking'}
+          disabled={!booking.formIsValid}
+        />
       </ButtonWrap>
-      :
-      <Button id="requestHolidayBtn" onClick={requestHoliday} label={booking.isWFH ? 'Add' : 'Request'} />
-  );
+    ) : (
+      <Button
+        id="requestHolidayBtn"
+        onClick={requestHoliday}
+        label={booking.form.isWFH.value ? 'Add' : 'Request'}
+        disabled={!booking.formIsValid}
+      />
+    );
 
-  const getTotalDays = () => {
-    if (booking.isWFH) { return '0'; }
-    return booking.duration / (booking.isHalfday ? 2 : 1);
+  const cssFormConfig = () => {
+    let cssClass = '';
+    if (booking.duration == 0) {
+      cssClass = 'workingFromHome';
+    } else if (booking.duration == 0.5) {
+      cssClass = 'bookingHalfDay';
+    }
+    return cssClass;
   };
 
+  let form = (
```

Comment:
Could use `const` here as it doesn't seem to ever change. But better again, this should be split out into its own component.

Reply:
Will update this. thanks